### PR TITLE
support slice assignment in ListConfig (#736)

### PR DESCRIPTION
**Pull request by pixelb**

### CODE REVIEW

**Comment by pixelb** [comment]
Need to double check negative indexes

File: omegaconf/listconfig.py [hunk 1]
```diff
@@ -241,7 +241,32 @@ def _set_at_index(self, index: Union[int, slice], value: Any) -> None:
 
     def __setitem__(self, index: Union[int, slice], value: Any) -> None:
         try:
-            self._set_at_index(index, value)
+            if isinstance(index, slice):
+                _ = iter(value)  # check iterable
+                indexes = range(*index.indices(len(self)))
+                if index.step is not None:
```
Comment (Jasha10) [inline omegaconf/listconfig.py:247]:
```suggestion
                if index.step not in (None, 1):
```
This makes `ListConfig.__setitem__` behave like `list.__setitem__` in the edge case where `step == 1`.

**Review by Jasha10 — APPROVED** [review]
Looks good!

File: tests/test_basic_ops_list.py [hunk 2]
```diff
@@ -845,6 +845,108 @@ def test_getitem_slice(sli: slice) -> None:
...
+            raises(ValueError),
+            id="extended-slice-length-mismatch2",
+        ),
+    ],
+)
+def test_setitem_slice(
+    lst: List[Any],
+    idx: slice,
+    value: Union[List[Any], Any],
+    expected: Union[List[Any], RaisesContext],
```
Comment (Jasha10) [inline tests/test_basic_ops_list.py:938]:
```suggestion
    expected: Union[List[Any], RaisesContext[Any]],
```
^ Hopefully this will fix the mypy error. Sorry about that.

**Comment by omry** [comment]
news fragment? :)
